docs/scripts/insert_timestamp.c

```diff
@@ -15,7 +18,35 @@ int main(int argc, char **argv)
    char buf[64];
    char const *version = "unknown";
 
-   time(&now);
+   char *source_date_epoch;
+   unsigned long long epoch;
+   char *endptr;
+
+   source_date_epoch = getenv("SOURCE_DATE_EPOCH");
+   if (source_date_epoch) {
+      errno = 0;
```

Comment:
Can we just use `sscanf`?

```c
uint64_t epoch;
char trailing;
int num_parsed = sscanf(source_date_epoch, "%"SCNu64"%c", &epoch, &trailing);
if (num_parsed != 1) // some error handling
```

Reply:
This code was simply copied from reproducible-builds.org with just tiny tweaks - unfortunately the site is down right now.

Reply:
Now the site seems to work again - This was simply copied from https://reproducible-builds.org/docs/source-date-epoch/  - See the C example a bit down on that page.